renderer/components/UI/Dropmenu.js

```diff
@@ -0,0 +1,372 @@
+  opacity: PropTypes.number,
+}
+
+DropmenuListScroller.defaultProps = {
+  opacity: 1,
+}
+
+const DropmenuListItem = ({ item }) => {
+  const { type, content, ...rest } = item
+
+  const renderContent = () => {
+    switch (type) {
+      case 'content':
+        return (
+          <Box as="li" css={{ position: 'sattic' }}>
+            <Box p={1}>{content}</Box>
+          </Box>
+        )
+      case 'bar':
+        return (
+          <Box as="li" css={{ position: 'sattic' }}>
+            <Bar variant="light" />
+          </Box>
+        )
+      default:
+        return (
+          <Box as="li" css={{ position: 'sattic' }}>
+            <DropmenuItem {...rest} />
+          </Box>
+        )
+    }
+  }
+
+  return renderContent()
+}
+
+const DropmenuList = React.forwardRef((props, forwardRef) => {
+  const { width } = useContext(MenuContext)
+  const ref = useRef(null)
+  const { scrollableHeight, isScrollbarVisible } = useComponentSize(ref)
+  const { y: scrollY = 0 } = useScroll(ref) || {}
+  const hasTopScroll = isScrollbarVisible && scrollY
+  const hasBottomScroll = isScrollbarVisible && scrollY < scrollableHeight
+
+  return (
+    <Card
+      ref={ref}
+      as="ul"
+      css={{
+        'max-height': '450px',
+        'overflow-x': 'hidden',
+        'overflow-y': 'auto',
+        '&::-webkit-scrollbar': {
+          display: 'none',
+        },
+      }}
+      p={0}
+      width={width}
+    >
+      <DropmenuListScroller direction="up" opacity={hasTopScroll ? 1 : 0}>
+        <AngleUp />
+      </DropmenuListScroller>
+      <Box>
+        <Box ref={forwardRef} {...props} m={isScrollbarVisible ? 2 : 0} />
+      </Box>
+      <DropmenuListScroller direction="down" opacity={hasBottomScroll ? 1 : 0}>
+        <AngleDown />
+      </DropmenuListScroller>
+    </Card>
+  )
+})
+
+DropmenuList.displayName = 'DropmenuList'
+
+const DropmenuListWithTheme = withTheme(DropmenuList)
+
+const DropmenuContent = ({ menuRef, items }) => {
+  return (
+    <DropmenuListWithTheme ref={menuRef}>
+      {items.map((item, index) => (
+        <DropmenuListItem key={item.id || index} item={item} />
+      ))}
+    </DropmenuListWithTheme>
+  )
+}
+
+DropmenuContent.propTypes = {
+  items: MenuItemsType,
+  menuRef: RefType,
+}
+
+const DropmenuSubmenu = props => {
+  const { top, left, ...rest } = props
+  return (
+    <Box
+      css={{
+        position: 'absolute',
+        'z-index': 10,
+        top: px(top),
+        left: px(left),
+      }}
+    >
+      <DropmenuContent {...rest} />
+    </Box>
+  )
+}
+
+DropmenuSubmenu.propTypes = {
+  left: PropTypes.number,
+  top: PropTypes.number,
+}
+
+DropmenuSubmenu.defaultProps = {
+  top: 0,
+  left: 0,
+}
+
+const DropmenuMenu = ({ menuRef, ...rest }) => {
+  const { justify } = useContext(MenuContext)
+  return (
+    <Box css={{ position: 'relative' }}>
+      <Box css={{ position: 'absolute', right: justify === 'right' ? 0 : null }}>
```

Comment:
Q: why `absolute` pos is used here for alignment and not `flexbox`

Reply:
This element needs to be absolutely positioned so that it sits above the page and existing page elements and not inline. With a flex or other inline element, opening the menu would cause all of the content below the menu to move. So since we are absolutely positioning anyway it makes sense to set the position at the same time rather than introducing another inner Flex just for the layout aspect.